Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ContestsController.cs

```diff
@@ -675,5 +734,29 @@ private void AddProblemGroupsToContest(Contest contest, int problemGroupsCount)
 
             return message;
         }
+
+        private int GetContestSubmissionsAverageRunTimeSeconds(Contest contest)
+        {
+            List<Submission> contestSubmissions = contest.ProblemGroups
+                                    .SelectMany(pg => pg.Problems)
+                                    .SelectMany(p => p.Submissions)
+                                    .ToList();
+
+            if (contestSubmissions.Any(s => s.StartedExecutionOn.HasValue))
+            {
+                return (int)contestSubmissions.Where(s => s.StartedExecutionOn.HasValue).Average(s => (s.ModifiedOn.Value - s.StartedExecutionOn.Value).TotalSeconds);
+            }
+            else
+            {
+                var contestSubmissionForCalc = contestSubmissions
+                    .Where(s => s.ModifiedOn.HasValue).ToList();
+
+                int countOfAllSubmissionTime = contestSubmissionForCalc
+                     .Sum(s => (int)s.ModifiedOn.Value.TimeOfDay.TotalSeconds);
+
+                return countOfAllSubmissionTime > 0 ?
+                    countOfAllSubmissionTime / contestSubmissionForCalc.Count() : 0;
```

Comment:
How you are able to get the runtime just from the `ModifiedOn`? Don't we need `CreatedOn` as well, to get the difference?